include/fs_utils.h

```diff
@@ -0,0 +1,31 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2019-2020  The dosbox-staging team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_FS_UTILS_H
+#define DOSBOX_FS_UTILS_H
+
+#include <string>
+
+// Checks if the given path corresponds to an existing file or directory.
+// Interface similar to C++17 std::filesystem::exists.
+
+bool fs_exists(const std::string &path) noexcept;
```

Comment:
@kcgen I decided to name this function `fs_exists` because C++17 `std::filesystem::exists`, but after looking at it for the second time, I am considering renaming it to `path_exists`. What do you think?

Reply:
I think `path_exists` is more accurate, and easier to remember. 

Also, given dosbox mounts filesystems (ISOs, IMGs, FATs, etc..) the concept of differentiating a directory versus a filesystem could be a function we want one day; similar to how `mountpoint` on Linux can differentiate a filesystem mount from a plain directory.